### ISSUE TITLE
Attachment Log - Upload Multiple Files

### ISSUE THREAD

**Issue by m-prodan** [bcgov/foi-flow#783]
* As an Intake/Flex/FOI Analyst or Ministry Coordinator
* I want to be able to upload multiple attachments
* so that I do not have to repeat the upload process when I have multiple attachments

**Assumptions & Scope**	
* This is an enhancement to functionality in #752
* File limit - 10 or 100 mb per upload
* no filtering required
* File types included, MVP: Excel (xls, xlsx, macro), pdf, image, word, email
* Unique file names are needed

What is IN scope?
The ability to upload multiple files to the attachment log

What is NOT in scope?
This functionality will NOT be extended to the replace file function in #782

**Acceptance Criteria**	

**Scenario 1: Drag and Drop multiple files**
* GIVEN a user is in the add attachment modal
* AND the user has already uploaded a file through this modal
* WHEN they drag an additional file into the drag and drop zone
* THEN the next file will show
* AND the user will be able to drag and drop another file

**Scenario 2: Add Multiple files using the "Add File" button**
* GIVEN a user is on the add attachment modal from the Attachment Page https://jacklyn808742.invisionapp.com/console/share/3W2O64AP5C/817813727
* AND a file has already been added to the modal
* WHEN you click on the "add file" button
* THEN the user will be able to select an additional file from their local machine to add to the upload file modal
* AND when the user selects a file and clicks "upload"
* THEN the file will appear in the modal
* AND the user will be able to upload additional files

**Dependencies? What is the impact of this dependency? (If so, link dependency in the ticket, make it visible in a team´s backlog)**
#752

**Validation Rules? (If yes, list here)**	

**Design**
@xxx - please link the Design here

**Definition of Ready**

1. [ ] Is there a well articulated User Story?
2. [ ] Is there Acceptance Criteria that covers all scenarios (happy/sad paths)?
3. [ ] If there is a user interface, is there a design? 
4. [ ] Does the user story need user research/validation?
5. [ ] Does this User Story needs stakeholder approval?
6. [ ] Design / Solution accepted by Product Owner
7. [ ] Is this user story small enough to be completed in a Sprint? Should it be split?
8. [ ] Are the dependencies known/ understood? (technical, business, regulatory/policy)
9. [ ] Has the story been estimated?

**Definition of Done**

1.    [ ] Passes developer unit tests
2.    [ ] Passes peer code review
3.    [ ] If there's a user interface, passes UX assurance
4.    [ ] Passes QA of Acceptance Criteria with verification in Dev and Test
5.	[ ] Confirm Test cases built and succeeding 
6.	[ ] No regression test failures
7.	[ ] Test coverage acceptable by Product Owner
8.	[ ] Ticket ready to be merged to master or story branch
9.	[ ] Developer to list Config changes/ Update documents and designs
10.	[ ] Can be demoed in Sprint Review 
11.	[ ] Tagged as part of a Release
12.   [ ] Feature flagged if required
13.  [ ] Change Management activities done?

**Comment by mpilchar** [comment]
Awesome job on this! One thing to flag while looking at it in test - I tried uploading a couple of high-res photos (larger files). There was a 10 second delay before the spinny load wheel appeared. So it was initially unclear if it worked or not. Same result in multiple browsers.

**Comment by mpilchar** [comment]
Hi there - I continued testing this today. Scenario 2 passed but I couldn't get Scenario 1 to work. The second file, which I dragged into the modal, automatically showed up as a download instead (.docx file). Another time it opened in a new browser tab (jpg file). I could not drag the file into the drag and drop zone when a current file was already there.

Steps to replicate: 
1) Login to test app as intake user on Chrome 
Click into a request 
Navigate to Attachments in left nav 
Click add attachment button 
Attach file from local drive using the button 
Drag a second file in from my local drive using drag and drop 

@divyav-aot @richard-aot @sumathi-thirumani-aot @abin-aot

**Comment by divyav-aot** [comment]
Hey @mpilchar @lmullane, Is it okay to create a bug for this issue and close this ticket? I will fix this bug after #782 US. Or, I will take this as priority and fix it. Please let me know your thoughts.

**Comment by mpilchar** [comment]
Hey @divyav-aot my thoughts would be to keep it open because AC is not yet met...I see a bug being more for an improvement or something that wasn't initially caught in the original AC. Recognizing that you've already started #782 I think the order/priority is up to you :)

**Comment by mpilchar** [comment]
@divyav-aot - Works beautifully, amazing work! Thanks for the fixes. Moving this to closed
